scripts: add generate-release-notes scripts

BethGriggs opened the pull request:
These scripts are used to generate release notes - detailed usage information is included in the README.md. The scripts have been migrated from https://github.com/BethGriggs/release-notes-prototype on the commit 906420d.

karianna commented on generate-release-notes/README.md:
```diff
@@ -0,0 +1,84 @@
+# OpenJDK Release Notes Tool
+
+This repository contains two scripts that can be used to generate OpenJDK notes. Typical usage will be to first generate the list of commits for the release using the `fetchCommitList.js` script and then run `fetchReleaseNotes.js` script, supplying the commit list as an input.
```
```suggestion
This repository contains two scripts that can be used to generate OpenJDK release notes. Typical usage will be to first generate the list of commits for the release using the `fetchCommitList.js` script and then run the `fetchReleaseNotes.js` script, supplying the commit list from `fetchCommitList.js` as an input.
```

BethGriggs replied:
It was intentional to provide some example output to share. I can delete (or reduce) if it is preferred.

karianna commented on generate-release-notes/README.md:
````diff
@@ -0,0 +1,84 @@
+]
+```
+
+Note that the same commit may reference multiple JDK issues.
````
```suggestion
*NOTE:* The same commit may reference multiple JDK issues.
```

karianna reviewed (changes requested):
Changes look largely good but not sure you meant to commit the output folder?

BethGriggs commented:
@karianna, I believe I've addressed your README comments. I also renamed the `output` directory to be `sample-output` for clarity. I still don't mind removing these - but I have found having some sample-output in the git tree useful to manually verify the impact of script changes, such as adding ProblemLists earlier - https://github.com/BethGriggs/release-notes-prototype/commit/575061ad8d9454079b21d1d5574482b13c1296a5.

I applied the fixes directly in https://github.com/BethGriggs/release-notes-prototype rather than here as I'd like this PR to be a clean switchover from the other repository at a specified commit. My plan is to archive the original repository as soon as this lands to avoid any divergence.

karianna commented:
@BethGriggs Safe for me to merge this?

sxa commented:
I'm going to merge this, update https://ci.adoptopenjdk.net/job/create_release_notes/configure to point to this repo, and run a test on it :-)
